arch/x86/kvm/pkvm/pkvm.c

```diff
@@ -568,14 +568,31 @@ static int pkvm_vcpu_put(int vm_handle, int vcpu_handle)
 	return ret;
 }
 
-static bool is_guest_vcpu_accessible(struct kvm_vcpu *vcpu)
+static bool is_guest_vcpu_accessible(struct kvm_vcpu *vcpu, enum pkvm_hc hc)
 {
 	/*
 	 * There is no isolation between non-protected VMs and the host, thus
-	 * all the PV interfaces are allowed for an npVM. For a pVM, by default
-	 * dis-allow all PV interfaces.
+	 * all the PV interfaces are allowed for an npVM.
 	 */
-	return !pkvm_is_protected_vcpu(vcpu);
+	if (!pkvm_is_protected_vcpu(vcpu))
+		return true;
+
+	switch (hc) {
+	/*
+	 * As the host needs to pre-configure the pVM's vCPU state for
+	 * booting, the protection for pVM is only enforced by the pKVM
+	 * hypervisor once the vCPU has started running.
```

Comment:
nit: would it look nicer to put this comment before `return !kvm_vcpu_has_run(vcpu)`, not here?

And same about the comment for the `return true` case.

Reply:
BTW another nit: would it look nicer to move the `return true` before this `return !kvm_vcpu_has_run(vcpu)` case? So that the order of cases is the order of increasing restriction: always allow everything for npVM, always allow some PV interfaces for pVM, sometimes allow some other PV interfaces for pVM, always deny any other interfaces for pVM. :)

Reply:
Yeah, sounds good to me.